mlir/lib/Dialect/Rock/utility/AmdArchDb.cpp

```diff
@@ -60,3 +60,32 @@ AmdArchInfo mlir::rock::lookupArchInfo(StringRef arch) {
                << arch << "\n";
   return gcnInfo;
 }
+
+GemmFeatures mlir::rock::AmdArchInfo::getDefaultFeatures(Type dataType) {
+  GemmFeatures theseFeatures = defaultFeatures;
+  bool isWmma = bitEnumContainsAll(theseFeatures, GemmFeatures::wmma);
+  if (isWmma) {
+    Type elementType = dataType;
+    if (auto vectorType = dataType.dyn_cast<VectorType>())
+      elementType = vectorType.getElementType();
+    else if (auto memrefType = dataType.dyn_cast<MemRefType>())
+      elementType = memrefType.getElementType();
+    if (!elementType.isF16() && !elementType.isBF16() &&
+        !elementType.isInteger(8)) {
+      theseFeatures = bitEnumClear(theseFeatures, GemmFeatures::wmma);
+    }
+  }
+  return theseFeatures;
+}
+
+GemmFeatures
+mlir::rock::AmdArchInfo::getDefaultFeatures(StringRef strDataType) {
+  GemmFeatures theseFeatures = defaultFeatures;
+  bool isWmma = bitEnumContainsAll(theseFeatures, GemmFeatures::wmma);
+  if (isWmma) {
+    if (strDataType != "f16" && strDataType != "bf16" && strDataType != "i8") {
+      theseFeatures = bitEnumClear(theseFeatures, GemmFeatures::wmma);
+    }
+  }
+  return theseFeatures;
+}
```

Comment:
This seems unnecessary? We have a `Conv2dGenerator::getDataType()` function that does type conversion. You should extract that to a header so this function doesn't have to operate on a string. Besides, having a duplicate compared to `getDefaultFeatures(Type dataType)` hard to maintain.

Reply:
You are 100% right! The reason I had two versions was because in the library we check the config "before" we build the module. But I didn't realize I could do the other way round. I just did, and got rid of the string version.